examples/streaming/knative-eventing/README.md

```diff
@@ -0,0 +1,387 @@
+      kind: SeldonDeployment
+      name: simple-iris-deployment
+
+```
+
+    Overwriting ./assets/seldon-knative-trigger.yaml
+
+
+Create this trigger file which will send all cloudevents of type `"seldon.<deploymentName>.request"`.
+
+
+```python
+!kubectl apply -f assets/seldon-knative-trigger.yaml
+```
+
+    trigger.eventing.knative.dev/seldon-eventing-sklearn-trigger unchanged
+
+
+CHeck that the trigger is working correctly (you should see "Ready: True"), together with the URL that will be reached.
+
+
+```python
+!kubectl get trigger 
+```
+
+    NAME                              READY   REASON   BROKER    SUBSCRIBER_URI                                                                                                          AGE
+    event-display                     True             default   http://event-display.default.svc.cluster.local/                                                                         3d5h
+    seldon-eventing-sklearn-trigger   True             default   http://istio-ingressgateway.istio-system.svc.cluster.local/seldon/default/simple-iris-deployment/api/v1.0/predictions   74m
+
+
+### Send a request to the KNative Eventing default broker
+
+To send requests we can do so by sending a curl command from a pod inside of the cluster.
+
+
+```python
+!kubectl run --quiet=true -it --rm curl --image=radial/busyboxplus:curl --restart=Never -- \
+    curl -v "default-broker.default.svc.cluster.local" \
+        -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
+        -H "Ce-specversion: 0.3" \
+        -H "Ce-Type: seldon.simple-iris-deployment.request" \
+        -H "Ce-Source: seldon.examples.streaming.curl" \
+        -H "Content-Type: application/json" \
+        -d '{"data": { "ndarray": [[1,2,3,4]]}}'
+```
+
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+
+
+### Check our model has received it
+
+We can do this by checking the logs (we can query the logs through the service name) and see that the request has been processed
+
+
+```python
+!kubectl logs svc/simple-iris-deployment-simple-iris-predictor simple-iris-model | tail -6
+```
+
+    2020-03-21 03:39:34,169 - SKLearnServer:predict:37 - INFO:  Calling predict_proba
+    2020-03-21 03:39:34,170 - werkzeug:_log:122 - INFO:  127.0.0.1 - - [21/Mar/2020 03:39:34] "[37mPOST /predict HTTP/1.1[0m" 200 -
+    2020-03-21 03:41:02,604 - SKLearnServer:predict:37 - INFO:  Calling predict_proba
+    2020-03-21 03:41:02,605 - werkzeug:_log:122 - INFO:  127.0.0.1 - - [21/Mar/2020 03:41:02] "[37mPOST /predict HTTP/1.1[0m" 200 -
+    2020-03-21 03:41:28,590 - SKLearnServer:predict:37 - INFO:  Calling predict_proba
+    2020-03-21 03:41:28,591 - werkzeug:_log:122 - INFO:  127.0.0.1 - - [21/Mar/2020 03:41:28] "[37mPOST /predict HTTP/1.1[0m" 200 -
+
+
+## Connect a source to listen to the results of the seldon model
+
+Our Seldon Model is producing results which are sent back to KNative.
```

Comment:
Does KNative store any of these results or does it just push them to the next trigger?

Reply:
The behaviour depends on the concept of the "Channel". The "chanel" is basically the component that stores all the messages. In this case, I am using the in-memory, channel, but in production you would use the Kafka Channel. However specific to your question, it indeed just pushes them to the registered observables (as it uses the observer pattern)